dot4gravity/src/lib.rs

```diff
@@ -1,8 +1,707 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use rand::Rng;
+
 #[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+mod tests;
+
+const BOARD_WIDTH: usize = 10;
+const BOARD_HEIGHT: usize = 10;
+const NUM_OF_PLAYERS: usize = 2;
+const NUM_OF_BOMBS_PER_PLAYER: usize = 3;
+const NUM_OF_BLOCKS: usize = 10;
+
+type Player = u32;
+
+/// Represents a cell of the board.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum Cell {
+    Empty,
+    Bomb([Option<Player>; NUM_OF_PLAYERS]),
+    Block,
+    Stone(Player),
+}
+
+impl Cell {
+    /// Tells if a cell is suitable for dropping a bomb.
+    fn is_valid_for_dropping_bomb(&self) -> bool {
+        match self {
+            Cell::Empty => true,
+            Cell::Bomb([Some(_), None]) => true,
+            _ => false,
+        }
+    }
+
+    /// Tells if a stone can traverse the cell.
+    fn is_travesable(&self) -> bool {
+        *self == Cell::Empty
+    }
+
+    /// Tells if a cell is of type 'bomb'
+    fn is_bomb(&self) -> bool {
+        match self {
+            Cell::Bomb([Some(_), None]) => true,
+            Cell::Bomb([Some(_), Some(_)]) => true,
+            _ => false,
+        }
+    }
+
+    /// Tells if a cell must be cleared when it's affected by an explosion.
+    fn is_explodable(&self) -> bool {
+        *self != Cell::Block
+    }
+}
+
+/// Coordinates for a cell in the board.
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct Coordinates {
+    row: usize,
+    col: usize,
+}
+
+impl Coordinates {
+    /// Tells if a cell is inside the board.
+    fn is_inside_board(&self) -> bool {
+        self.row < BOARD_WIDTH && self.col < BOARD_HEIGHT
+    }
+
+    /// Tells if a cell is in the oposite of a side.
+    fn is_opposite_cell(&self, side: Side) -> bool {
+        match side {
+            Side::North => self.row == BOARD_HEIGHT - 1,
+            Side::East => self.col == 0,
+            Side::South => self.row == 0,
+            Side::West => self.col == BOARD_WIDTH - 1,
+        }
+    }
+}
+
+/// Sides of the board from which a player can drop a stone.
+#[derive(Clone, Debug)]
+enum Side {
+    North,
+    East,
+    South,
+    West,
+}
+
+#[derive(Clone, Debug)]
+struct Board {
+    cells: [[Cell; BOARD_WIDTH]; BOARD_HEIGHT],
+}
+
+impl Board {
+    pub fn new() -> Board {
+        Board {
+            cells: [[Cell::Empty; BOARD_WIDTH]; BOARD_HEIGHT],
+        }
+    }
+
+    fn get_cell(&self, position: &Coordinates) -> Cell {
+        self.cells[position.row][position.col]
+    }
+
+    fn change_cell(&mut self, position: Coordinates, cell: Cell) {
```

Comment:
Let's rename this to `update_cell`:
```suggestion
    fn update_cell(&mut self, position: Coordinates, cell: Cell) {
```

Reply:
changed